homeassistant/components/homee/cover.py

```diff
@@ -0,0 +1,258 @@
+    open_close = None
+
+    # We assume, that no device has UP_DOWN and OPEN_CLOSE, but only one of them.
+    if node.has_attribute(AttributeType.UP_DOWN) or node.has_attribute(
+        AttributeType.OPEN_CLOSE
+    ):
+        if node.has_attribute(AttributeType.UP_DOWN):
+            open_close = AttributeType.UP_DOWN
+        else:
+            open_close = AttributeType.OPEN_CLOSE
+
+        if node.get_attribute(open_close).editable:
+            features |= (
+                CoverEntityFeature.OPEN
+                | CoverEntityFeature.CLOSE
+                | CoverEntityFeature.STOP
+            )
+
+    # Check for up/down position settable.
+    if node.has_attribute(AttributeType.POSITION):
+        if node.get_attribute(AttributeType.POSITION).editable:
+            features |= CoverEntityFeature.SET_POSITION
+
+    if node.has_attribute(AttributeType.SLAT_ROTATION_IMPULSE):
+        features |= CoverEntityFeature.OPEN_TILT | CoverEntityFeature.CLOSE_TILT
+
+    if node.has_attribute(AttributeType.SHUTTER_SLAT_POSITION):
+        features |= CoverEntityFeature.SET_TILT_POSITION
+
+    return features, open_close
+
+
+def get_device_class(node: HomeeNode) -> CoverDeviceClass | None:
+    """Determine the device class a homee node based on the node profile."""
+    COVER_DEVICE_PROFILES = {
+        NodeProfile.GARAGE_DOOR_OPERATOR: CoverDeviceClass.GARAGE,
+        NodeProfile.SHUTTER_POSITION_SWITCH: CoverDeviceClass.SHUTTER,
+    }
+
+    return COVER_DEVICE_PROFILES.get(node.profile, None)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: HomeeConfigEntry,
+    async_add_devices: AddEntitiesCallback,
+) -> None:
+    """Add the homee platform for the cover integration."""
+
+    devices: list[HomeeCover] = []
+    nodes = config_entry.runtime_data.nodes
+    devices.extend(
+        HomeeCover(node, config_entry) for node in nodes if is_cover_node(node)
+    )
+
+    async_add_devices(
+        HomeeCover(node, config_entry)
+        for node in config_entry.runtime_data.nodes
+        if is_cover_node(node)
+    )
+
+
+def is_cover_node(node: HomeeNode) -> bool:
+    """Determine if a node is controllable as a homee cover based on its profile and attributes."""
+    return node.profile in [
+        NodeProfile.ELECTRIC_MOTOR_METERING_SWITCH,
+        NodeProfile.ELECTRIC_MOTOR_METERING_SWITCH_WITHOUT_SLAT_POSITION,
+        NodeProfile.GARAGE_DOOR_OPERATOR,
+        NodeProfile.SHUTTER_POSITION_SWITCH,
+    ]
+
+
+class HomeeCover(HomeeNodeEntity, CoverEntity):
+    """Representation of a homee cover device."""
+
+    def __init__(self, node: HomeeNode, entry: HomeeConfigEntry) -> None:
+        """Initialize a homee cover entity."""
+        super().__init__(node, entry)
+        self._attr_supported_features, self._open_close_attribute = get_cover_features(
+            self
+        )
+        self._device_class = get_device_class(node)
+        self._attr_name = None
```

Comment:
Can be set outside of the constructor

Reply:
_attr_name = none?
I see, every integration sets it in the constructor...

Reply:
When its static its usually set outside

Reply:
OK, then I just managed to look at the ones who don't have it static. :-)